src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -102,9 +102,9 @@ public async Task<HttpResponse> SendRequestAsync(
                 timeoutException = exception;
             }
 
-            while (_retryPolicy.pauseForRetry(response, timeoutException, retryCount))
+            while (await _retryPolicy.PauseForRetryAsync(response, timeoutException, retryCount, logger).ConfigureAwait(false))
             {
-                logger.Warning($"Retry condition met. Retry count: {retryCount++} after waiting {_retryPolicy.DelayInMilliseconds}ms.");
+                retryCount++;
```

Comment:
we now increment once here? and the previous log line that printed the delay is gone? is it now logged inside `PauseForRetryAsync`

Reply:
correct

Reply:
the delay is calculated inside of each retry policy